ru_RU/howToGetBeanFromSpringContext.q.md

```diff
@@ -0,0 +1,6 @@
+Как достать бин из спринг контекста
+
+* ApplicationContext.getBeanDefinitionNames() - вернет имена всех правильно загруженных в контекст бинов.
+* getBean("name") - вернет бин с указанным именем, если такой загружен в контекст.
+* ApplicationContext.getBeansOfType() — с помощью этого метода мы можем загрузить все бины определенного типа. Он вернет Map<String, Object> — где ключ это имя компонента, а Object — фактический объект компонента.
+* ApplicationContext.getBeansWithAnnotation() - с помощью этого метода мы можем замапить все бины определенного типа, т.е. Service, Controller и другие аннотации с помощью которых создаются бины
```

Comment:
```java
getBean(Name.class)
```

bean could be injected into component property using @Autowired annotation on getter or constructor
Spring assume if you declare constructor with parameters for initialize  private fields that fields should be injected from context